addons/pos_loyalty/static/src/overrides/models/pos_order.js

```diff
@@ -1388,4 +1388,33 @@ patch(PosOrder.prototype, {
             return super.removeOrderline(lineToRemove);
         }
     },
+    getSortedOrderlines() {
+        const lines = super.getSortedOrderlines();
+        if (this.config.orderlines_sequence_in_cart_by_category && this.lines.length) {
+            const rewardLines = [];
+            const resultLines = [];
+
+            lines.forEach((line) => {
+                if (line.is_reward_line) {
+                    rewardLines.push(line);
+                } else {
+                    resultLines.push(line);
+                }
+            });
+
+            rewardLines.forEach((line) => {
+                if (line.reward_id.reward_type === "discount") {
+                    resultLines.splice(resultLines.length, 0, line);
+                } else if (line.reward_id.reward_type === "product") {
+                    const rewardProductIndex = resultLines.findIndex(
+                        (rewardLine) =>
+                            line.reward_id?.reward_product_id?.id === rewardLine.product_id.id
+                    );
+                    resultLines.splice(rewardProductIndex + 1, 0, line);
+                }
+            });
```

Comment:
Yes, as per MOBT's suggestion, we need to sort the reward line along with other categories if the reward line is of product type. Otherwise, add it at the end, such as for discount.